lib/iex/lib/iex/info.ex

```diff
@@ -290,6 +290,33 @@ defimpl IEx.Info, for: Reference do
   end
 end
 
+defimpl IEx.Info, for: [Date, Time, NaiveDateTime] do
+  {sigil, repr} =
+    case @for do
+      Date -> {"D", "date"}
+      Time -> {"T", "time"}
+      NaiveDateTime -> {"N", ~S{"naive" datetime (that is, a datetime without a timezone)}}
+    end
+
+  def info(value) do
+    desc = """
+    This is a struct representing a #{unquote(repr)}. It is commonly represented
+    using the `~#{unquote(sigil)}` sigil syntax, that is defined in the `Kernel.sigil_#{unquote(sigil)}/2` macro.
+    """
+    ["Data type": inspect(@for),
+     "Description": desc,
+     "Raw representation": raw_inspect(value),
+     "Reference modules": inspect(@for) <> ", Calendar, Map"]
+  end
+
+  defp raw_inspect(value) do
+    value
+    |> Inspect.Any.inspect(%Inspect.Opts{})
```

Comment:
Quoting @michalmuskala a few comments above:

> This gives raw map %{__struct__: Foo, ...} and I don't think we need that here. A plain struct works better, imho.

If we use `structs: false`, we get the map representation, but here we want to still show `%Date{...}` because users may have called `~D()`. I think that's the idea at least, I'm fine with showing the raw map representation as well personally.